http-server/src/main/java/io/micronaut/http/server/OptionsFilter.java

```diff
@@ -68,13 +67,15 @@ public HttpResponse<?> filterRequest(HttpRequest<?> request) {
         if (hasOptionsRouteMatch(request)) {
             return null; // proceed
         }
-        MutableHttpResponse<?> mutableHttpResponse = HttpResponse.status(HttpStatus.OK);
-        router.findAny(request.getUri().toString(), request)
-            .map(UriRouteMatch::getHttpMethod)
-            .map(HttpMethod::toString)
-            .forEach(allow -> mutableHttpResponse.header(HttpHeaders.ALLOW, allow));
-        mutableHttpResponse.header(HttpHeaders.ALLOW, HttpMethod.OPTIONS.toString());
-        return mutableHttpResponse;
+        if (HttpStatus.METHOD_NOT_ALLOWED.equals(response.getStatus())) {
```

Comment:
at this point might as well move the entire logic into RequestLifecycle or wherever METHOD_NOT_ALLOWED is thrown...

Reply:
@yawkat Yes, I had that same thought. It happens in [RequestLifecycle#onRouteMiss](https://github.com/micronaut-projects/micronaut-core/blob/622cb9698249db85837194396d07bf35147ba9fa/http-server/src/main/java/io/micronaut/http/server/RequestLifecycle.java#L305).

My main concern, and reason for not doing that yet, is that `onRouteMiss` executes before `runWithFilters` and the original intention of this filter to avoid conflicting with the CORS filter. 

That said, it occurs to me now that the `onRouteMiss` logic must be getting executed anyway (and the `Allow` header getting populated) even in the case of CORS being enabled. Thus in a Servlet environment, we are likely sending the `Allow` header even in response to a CORS pre-flight request (alongside all the other headers that CORS adds) because of the same issue of not being able to create a new ServletResponse in the middle of request processing.